ethereum/spec/src/test/java/tech/pegasys/teku/spec/datastructures/state/beaconstate/common/AbstractBeaconStateTest.java

```diff
@@ -71,16 +71,18 @@ public void simpleMutableBeaconStateTest() {
   }
 
   @SuppressWarnings("unchecked")
+  @Test
   public void equals_shouldReturnTrue() {
     final T state = randomState();
-    final T stateCopy = (T) state.updated(__ -> {});
+    final Object stateCopy = state.getSchema().createFromBackingNode(state.getBackingNode());
 
     assertThat(state == stateCopy).isFalse();
     assertThat(state).isEqualTo(stateCopy);
     assertThat(state.hashCode()).isEqualTo(stateCopy.hashCode());
   }
 
   @SuppressWarnings("unchecked")
+  @Test
   public void equals_shouldReturnFalse() {
```

Comment:
Nice catch. I guess these weren't being tested before.

Reply:
Yeah I was pleased to catch that. They aren't particularly important tests but still nice to fix.